wen-restart/src/wen_restart.rs

```diff
@@ -264,12 +276,16 @@ pub(crate) fn find_heaviest_fork(
         .map(|(slot, _)| *slot)
         .collect::<Vec<Slot>>();
     slots.sort();
+
+    // The heaviest slot we selected will always be the last of the slots list, or root if the list is empty.
+    let heaviest_fork_slot = slots.last().map_or(root_slot, |x| *x);
+
     let mut expected_parent = root_slot;
-    for slot in slots {
+    for slot in &slots {
```

Comment:
Suppose `slots = [s_1, s_2, ... , s_n]`. This loop is enforcing the case that `s_i` is the parent of `s_i+1` right?

If so, is it guaranteed that `slots` will contain only one fork? If not, the following fork graph could mess things up
```
1 - 3 - 5
  \ 2 - 4
```
After sort, our list would be `[1, 2, 3, 4, 5]` which obviously doesn't reflect the parent-child relationships

Reply:
That is a good question. In this specific case we can guarantee there is only one fork, and it's verified in find_heaviest_fork(). The correctness proof is in the yet-to-be-final-reviewed SIMD: https://github.com/solana-foundation/solana-improvement-documents/pull/46 (line 247-282).

Reply:
> and it's verified in find_heaviest_fork(). The correctness proof is in the yet-to-be-final-reviewed SIMD

Ahh ok, so to confirm my understanding, we should only have a single fork coming in (per the correctness proof), but `find_heaviest_fork()` is the sanity check to ensure that is the case before we continue?

Reply:
Yes, if the check fails we can't proceed safely, must abort the restart and wait for manual inspection.